cmd/login/auth.go

```diff
@@ -222,19 +225,24 @@ func switchContext(ctx context.Context, k8sConfigAccess clientcmd.ConfigAccess,
 			if err != nil {
 				return microerror.Mask(tokenRenewalFailedError)
 			}
+			hasNewTokens = authProvider.Config[RefreshToken] != rToken || authProvider.Config[IDToken] != idToken
 			authProvider.Config[RefreshToken] = rToken
 			authProvider.Config[IDToken] = idToken
 		}
 	} else if authType == kubeconfig.AuthTypeUnknown {
 		return microerror.Maskf(incorrectConfigurationError, "There is no authentication configuration for the '%s' context", newContextName)
 	}
-	if switchContext {
+
+	if switchContext && !isContextAlreadySelected {
 		config.CurrentContext = newContextName
+		hasNewContext = true
 	}
 
-	err = clientcmd.ModifyConfig(k8sConfigAccess, *config, true)
-	if err != nil {
-		return microerror.Mask(err)
+	if hasNewContext || hasNewTokens {
```

Comment:
We detect actual changes and only update the kubeconfig file in case there are some